homeassistant/components/cloud/backup.py

```diff
@@ -120,6 +121,8 @@ async def async_upload_backup(
         """
         if not backup.protected:
             raise BackupAgentError("Cloud backups must be protected")
+        if self._cloud.subscription_expired:
+            raise BackupAgentError("Cloud subscription has expired")
```

Comment:
A future improvement could be to create a repair issue asking the user to renew their subscription.